### PR TITLE
refactor: webapp ci

### PR DESCRIPTION
**Pull request by GabenGar**
- added `.dockerignore` to webapp so it wouldn't dump the local build artifacts and `node_modules` folder into the image
- changed dockerfile so it would actually install with yarn and the package cache wouldn't bust on every source code change

### CODE REVIEW

**Comment by GabenGar** [comment]
@horahoradev 
For some reason yarn couldn't fetch the packages when ran from the docker stack. Installs fine from the local setup though.
Does horahora use some kind of proxy and can you check if the webapp image builds for you?

**Comment by horahoradev** [comment]
There wouldn't be any proxy for yarn, no. It does build fine on the CI tests.

**Comment by horahoradev** [comment]
I don't see any issues with this... but I'll merge it tomorrow morning; apologies for the delay. Thank you for your contribution!

File: webapp/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 
 WORKDIR /webapp
 
-COPY . .
+# https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy
+COPY ["package.json", "yarn.lock", "/webapp"]
+
+RUN yarn install
 
-RUN cd /webapp && \
```
Comment (horahoradev) [inline webapp/Dockerfile:9]: so this way we won't have to reinstall deps when we change the src (lol). much better

File: webapp/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+**/.dockerignore
+**/.env
+**/.git
+**/.gitignore
+**/.vscode
+**/Dockerfile*
```
Comment (horahoradev) [inline webapp/.dockerignore:6]: oh, interesting, I hadn't heard of a dockerignore file

**Review by horahoradev — APPROVED** [review]
WOW!

**Comment by horahoradev** [comment]
merged, thank you!

**Pull request merged**